File: LinkedList/Python/Insertion/insertBEG.py
class node:
    def __init__(self,data):
        self.data=data
        self.next=None
class LinkedList:
    def __init__(self):
        self.head=None

    #insert at beggining of a Linked List.
    def insertbeg(self, data):
        newnode=node(data)
        ptr=self.head

        if ptr is None:
            self.head=newnode
        else:
            newnode.next=self.head
            self.head=newnode

    def display(self):
        ptr=self.head
        while(ptr != None):
            print(ptr.data,end=" -> ")
            ptr=ptr.next
        print ("NULL. \n")

head=LinkedList()
head.insertbeg(1)
head.insertbeg(2)
head.insertbeg(3)
head.insertbeg(4)
head.insertbeg(5)
head.display()

File: LinkedList/C/Insertion/insertBEG.c
#include<stdio.h>
#include<stdlib.h>
struct node{
    int data;
    struct node *next;
}nn;

struct node *addbeg(struct node *head,int data){
    struct node *ptr;
    struct node *newnode=(struct node*)malloc(sizeof(nn));
    newnode->data = data;
    newnode->next = NULL;
    ptr=head;
    if(head==NULL)
    head=newnode;
    else{
        newnode->next=head;
        head=newnode;
    }
    return head;
}

void display(struct node *head){
    struct node *ptr=head;
    while(ptr != NULL)
    {
        printf("%d -> ",ptr->data);
        ptr=ptr->next;
    }
    printf("NULL. \n");
}

int main()
{
    struct node *head=(struct node*)malloc(sizeof(nn));
    head->data=1;
    head->next=NULL;

    head = addbeg(head, 2);
    head = addbeg(head, 3);
    head = addbeg(head, 4);
    head = addbeg(head, 5);
    display(head);

    return 0;
}